src/java.base/share/classes/sun/net/www/http/KeepAliveCache.java

```diff
@@ -180,38 +208,34 @@ private static int getUserKeepAlive(boolean isProxy) {
         return isProxy ? userKeepAliveProxy : userKeepAliveServer;
     }
 
-    /* remove an obsolete HttpClient from its VectorCache */
-    public synchronized void remove(HttpClient h, Object obj) {
-        KeepAliveKey key = new KeepAliveKey(h.url, obj);
-        ClientVector v = super.get(key);
-        if (v != null) {
-            v.remove(h);
-            if (v.isEmpty()) {
-                removeVector(key);
-            }
-        }
-    }
-
-    /* called by a clientVector thread when all its connections have timed out
+    /*
+     * called by a clientVector thread when all its connections have timed out
      * and that vector of connections should be removed.
      */
-    synchronized void removeVector(KeepAliveKey k) {
+    private void removeVector(KeepAliveKey k) {
+        assert cacheLock.isHeldByCurrentThread();
```

Comment:
Not part of the original changeset. Please keep the port as close as possible to the JDK 17 version. Caveats such as JDK-8229867 apply.